apps/sms/views/shared/test/unit/navigation_test.js

```diff
@@ -657,6 +657,8 @@ suite('navigation >', function() {
 
       window.Navigation = NavigationFactory(fakeWindow);
       navigator.mozHasPendingMessage.withArgs('notification').returns(false);
+
+      Navigation.setReady();
```

Comment:
What we need is to have a test that's failing before your patch and working with your patch.

so it should be possible to fake `document.readyState` to `interactive`, call toPanel or init, and do a `.then(done, done)` at the end (without sending "load" event but calling setReady). This should fail on master and work with your patch.

What do you think ?